crates/dreammaker/src/preprocessor.rs

```diff
@@ -952,6 +967,18 @@ impl<'ctx> Preprocessor<'ctx> {
                         expect_token!((text) = Token::String(text));
                         self.context.register_error(DMError::new(self.last_input_loc, format!("#{} {}", ident, text)));
                     }
+                    "pragma" if disabled => {}
+                    "pragma" => {
+                        expect_token!((text) = Token::String(text));
```

Comment:
You probably mean Ident here, not String

Reply:
I changed it, but are you sure? #warn uses String() and its input isn't "'d at all, it's just the text after the #warn

Reply:
It's stupid, but there's a hack in lexer.rs for this: `if ident == "warn" || ident == "error" {`

Reply:
in fucking tears

Reply:
should I add to that or just keep as is

Reply:
As written is fine. The reason is that #warn and #error want to treat the whole rest of the line as one big string. #pragma is (for now...) followed by ordinary tokens.

Reply:
alright. is there anything else you want me to do here?